versions-common/src/main/java/org/codehaus/mojo/versions/recording/ChangeRecorderXML.java

```diff
@@ -72,31 +82,40 @@ public ChangeRecorderXML()
     }
 
     @Override
-    public final void recordUpdate( final String kind, final String groupId, final String artifactId,
-                                    final String oldVersion, final String newVersion )
+    public final void recordUpdate( ChangeRecord changeRecord )
     {
-        Objects.requireNonNull( kind, "kind" );
-        Objects.requireNonNull( groupId, "groupId" );
-        Objects.requireNonNull( artifactId, "artifactId" );
-        Objects.requireNonNull( oldVersion, "oldVersion" );
-        Objects.requireNonNull( newVersion, "newVersion" );
-
         final Element update = this.document.createElementNS( CHANGES_NAMESPACE, "update" );
```

Comment:
How about using Modello? 😄

Reply:
I saw it.
Yes Modello should be used here for consistencies - now this report is undocumented.
Out of scope for this change.

Reply:
Fair enough.